src/Agent.Worker/Build/GitCommandManager.cs

```diff
@@ -232,8 +232,8 @@ public async Task<int> GitFetch(IExecutionContext context, string repositoryPath
                 refSpec = refSpec.Where(r => !string.IsNullOrEmpty(r)).ToList();
             }
 
-            // insert prune-tags if knob is false to sync tags with the remote
-            string pruneTags = AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
+            // insert prune-tags if DisableFetchPruneTags knob is false and Git version is above 2.17
+            string pruneTags = !EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) ? string.Empty : AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean() ? string.Empty : "--prune-tags";
```

Comment:
Here is exactly the same:

```suggestion
            string pruneTags = string.Empty;
            if (EnsureGitVersion(new Version(2, 17), throwOnNotMatch: false) && !AgentKnobs.DisableFetchPruneTags.GetValue(context).AsBoolean())
            {
                pruneTags = "--prune-tags";
            }
```

Reply:
Yes, seems good